roles/confluent.kafka_connect/library/confluent_connectors.py

```diff
@@ -0,0 +1,172 @@
+  active_connectors: [{"name": "test-6-sink", "config": { .../... }},{"name": "test-5-sink", "config": { .../... }}]
+'''
+
+RETURN = '''
+message:
+    description: The output message that the module generates
+    type: str
+    returned: always
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+
+
+def get_current_connectors(connect_url):
+    try:
+        res = open_url(connect_url)
+        return json.loads(res.read())
+    except urllib_error.HTTPError as e:
+        if e.code != 404:
+            raise
+        return []
+
+def remove_connector(connect_url, name):
+    url = "{}/{}".format(connect_url, name)
+    r = open_url(method='DELETE', url=url)
+    return r.getcode() == 200
+
+def install_new_connector(connect_url, name, config):
+    data = json.dumps({'name': name, 'config': config})
+    headers = {'Content-Type': 'application/json'}
+    r = open_url(method='POST', url=connect_url, data=data, headers=headers)
+    return r.getcode() in (200, 201, 409)
+
+def update_existing_connector(connect_url, name, config):
+    url = "{}/{}/config".format(connect_url, name)
+    restart_url = "{}/{}/restart".format(connect_url, name)
+
+    res = open_url(url)
+    current_config = json.loads(res.read())
+
+    existing_config = config.copy()
+    existing_config.update({'name': name})
+
+    if current_config == existing_config:
+        return False
+
+    data = json.dumps(config)
+    headers = {'Content-Type': 'application/json'}
+    r = open_url(method='PUT', url=url, data=data, headers=headers)
+
+    changed = r.getcode() in (200, 201, 409)
+
+    r = open_url(method='POST', url=restart_url)
+    if r.getcode() not in (200, 204, 409):
+        raise Exception("Connector {} failed to restart after a configuration update. {}".format(name, r.msg))
+
+    return changed
+
+
+def run_module():
+    module_args = dict(
+        connect_url=dict(type='str', required=True),
+        active_connectors=dict(type='list', required=True),
+    )
+
+    result = dict(changed=False, message='')
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    #
+    # module action:
+    # - make a diff of existing (current) vs kept (active) connectors and removes the un-kept ones
+    # - update the connector if it already exists, otherwise create a new one
+    #
+    # note: using the logic below because PUT /connectors/<name>/config has proven to be unreliable
+    # when the connector doesn't exist
+    #
+    result['changed'] = False
+    output_messages = []
+    try:
+        current_connector_names = get_current_connectors(connect_url=module.params['connect_url'])
+        active_connector_names = [c['name'] for c in module.params['active_connectors']]
```

Comment:
No need to allocate a list here

```suggestion
        active_connector_names = (c['name'] for c in module.params['active_connectors'])
```